components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/APIResourceManagementDAOImpl.java

```diff
@@ -0,0 +1,944 @@
+import org.wso2.carbon.identity.api.resource.mgt.constant.SQLConstants;
+import org.wso2.carbon.identity.api.resource.mgt.dao.APIResourceManagementDAO;
+import org.wso2.carbon.identity.api.resource.mgt.model.FilterQueryBuilder;
+import org.wso2.carbon.identity.api.resource.mgt.util.APIResourceManagementUtil;
+import org.wso2.carbon.identity.application.common.model.APIResource;
+import org.wso2.carbon.identity.application.common.model.ApplicationBasicInfo;
+import org.wso2.carbon.identity.application.common.model.Scope;
+import org.wso2.carbon.identity.core.model.ExpressionNode;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.api.resource.mgt.constant.APIResourceManagementConstants.AFTER;
+import static org.wso2.carbon.identity.api.resource.mgt.constant.APIResourceManagementConstants.BEFORE;
+
+/**
+ * This class implements the {@link APIResourceManagementDAO} interface.
+ */
+public class APIResourceManagementDAOImpl implements APIResourceManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(APIResourceManagementDAOImpl.class);
+
+    @Override
+    public List<APIResource> getAPIResources(Integer limit, Integer tenantId, String sortOrder,
+                                             List<ExpressionNode> expressionNodes)
+            throws APIResourceMgtException {
+
+        return getAPIResourcesList(limit, tenantId, sortOrder, expressionNodes);
+    }
+
+    @Override
+    public Integer getAPIResourcesCount(Integer tenantId, List<ExpressionNode> expressionNodes)
+            throws APIResourceMgtException {
+
+        // Remove the after and before attributes from the expression nodes list,
+        // since we are not considering them when getting the total count.
+        List<ExpressionNode> expressionNodesCopy = new ArrayList<>(expressionNodes);
+        if (!expressionNodes.isEmpty()) {
+            expressionNodesCopy.removeIf(expressionNode -> AFTER.equals(expressionNode.getAttributeValue()) ||
+                    BEFORE.equals(expressionNode.getAttributeValue()));
+        }
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNodesCopy, filterQueryBuilder, false);
+
+        Map<Integer, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        String sqlStmt;
+        String getAPIResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL;
+
+        sqlStmt = SQLConstants.GET_API_RESOURCES_COUNT + filterQueryBuilder.getFilterQuery() +
+                getAPIResourcesCountSqlStmtTail;
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
```

Comment:
use try-with-resources to properly close the connections, prepared statements and result sets.

Reply:
Check all the methods in the DAO.